sdk/core/ts-http-runtime/test/util/delay.spec.ts

```diff
@@ -1,86 +1,51 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { describe, it, assert, vi, afterEach } from "vitest";
-import { calculateRetryDelay, delay } from "../../src/util/delay.js";
-
-describe("delay", function () {
-  afterEach(function () {
-    vi.useRealTimers();
+import { describe, it, assert } from "vitest";
+import { calculateRetryDelay } from "../../src/util/delay.js";
+
+describe("calculateRetryDelay", function () {
+  it("should calculate the correct delay for the first retry attempt", function () {
+    const retryAttempt = 1;
+    const config = {
+      retryDelayInMs: 1000,
+      maxRetryDelayInMs: 10000,
+    };
+    const result = calculateRetryDelay(retryAttempt, config);
+    assert.isAtLeast(result.retryAfterInMs, config.retryDelayInMs);
+    assert.isAtMost(result.retryAfterInMs, config.retryDelayInMs * 2);
   });
 
-  it("should resolve after the given number of ms", async function () {
-    const clock = vi.useFakeTimers();
-    const delayTime = 2500;
-    const delayPromise = delay(delayTime);
-    await clock.advanceTimersByTimeAsync(delayTime);
-    assert.strictEqual(clock.getTimerCount(), 0);
-    clock.useRealTimers();
-    // should be resolved, so we can await it and it will resolve next tick
-    await delayPromise;
+  it("should calculate the correct delay for multiple retry attempts", function () {
+    const retryAttempt = 3;
+    const config = {
+      retryDelayInMs: 1000,
+      maxRetryDelayInMs: 10000,
+    };
+    const result = calculateRetryDelay(retryAttempt, config);
+    const expectedMin = config.retryDelayInMs * Math.pow(2, retryAttempt - 1);
+    const expectedMax = config.retryDelayInMs * Math.pow(2, retryAttempt);
+    assert.isAtLeast(result.retryAfterInMs, expectedMin);
+    assert.isAtMost(result.retryAfterInMs, Math.min(expectedMax, config.maxRetryDelayInMs));
   });
 
-  it("should return when the abort signal is called", async function () {
```

Comment:
Is this test moved anywhere or deleted (and if so, mind helping me understand why)?

Reply:
I removed `delay` from the unbranded package entirely because it's not actually being used anywhere there. That means this test doesn't actually have a function to test anymore, so I deleted it along with all the other test for `delay` that was in this file. The tests still exist in `core-util` [here](https://github.com/Azure/azure-sdk-for-js/blob/9695688fc5ed91fe5d12654460551757ab4a9078/sdk/core/core-util/test/public/delay.spec.ts#L23) since that's where `delay` now lives. Does that make sense?